src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -337,6 +338,14 @@ public function get_items( $request ) {
 			}
 		}
 
+		/*
+		 * Honor the original REST API `post__in` behavior. Don't prepend sticky posts
+		 * when `post__in` has been specified.
+		 */
+		if ( ! empty( $args['post__in'] ) ) {
+			unset( $args['ignore_sticky_posts'] );
+		}
```

Comment:
Props to unit tests for catching this, but should we keep this behavior backward compatible?

cc @spacedmonkey, @TimothyBJacobs

Reply:
My $0.02, if I understand to question correctly, is that I would expect that if someone specifically set a `post__in` param, that WP wouldn't _also_ return sticky posts. So keeping the current behavior would be correct.

I'm not seeing a test case that specifically covers not including sticky posts when `include` is used. Should we add that too?

Reply:
That’s correct.

It was a different test that caught this, but I’ll add separate test case.